tsl/test/sql/data_node.sql

```diff
@@ -55,11 +55,20 @@ SELECT * FROM add_data_node('data_node_2', host => 'localhost', database => :'DN
 
 SELECT * FROM add_data_node('data_node_3', host => 'localhost', database => :'DN_DBNAME_3');
 
--- Test altering server command is blocked
+
+-- Altering the host, dbname, and port should work via ALTER SERVER
+BEGIN;
+ALTER SERVER data_node_3 OPTIONS (SET host 'data_node_3', SET dbname 'new_db_name', SET port '9999');
```

Comment:
Since we are changing host name here, I believe it would make sense to test how we (and if we) clean up our connection cache here. What if we openned connection to the previous host and then change the host name?

Reply:
There's already a callback to invalidate a connection when the foreign server changes (see the connection cache code). That should already be tested.